src/adm.rs

```diff
@@ -1,43 +1,279 @@
 use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashMap};
 use url::Url;
 
-use crate::error::HandlerError;
+use crate::error::{HandlerError, HandlerErrorKind, HandlerResult};
+use crate::server::location::LocationResult;
+use crate::server::ServerState;
+use crate::settings::Settings;
+use crate::tags::Tags;
+use crate::web::middleware::sentry as l_sentry;
+//use crate::server::img_storage;
+
+pub(crate) const DEFAULT: &str = "DEFAULT";
 
 #[derive(Debug, Deserialize, Serialize)]
 pub struct AdmTileResponse {
     pub tiles: Vec<AdmTile>,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+/// Filter criteria for adm Tiles
+#[derive(Default, Clone, Debug)]
+pub struct AdmFilter {
+    /// list of allowed base host strings.
+    pub filter_set: BTreeMap<String, AdmAdvertiserFilterSettings>,
+}
+
+/// The AdmAdvertiserFilterSettings contain the settings for the various
+/// ADM provided partners. These are specified as a JSON formatted hash
+/// that contains the components. A special "DEFAULT" setting provides
+/// information that may be used as a DEFAULT, or commonly appearing set
+/// of data.
+/// See `impl From<Settings>` for details of the structure.
+#[derive(Clone, Debug, Deserialize, Default)]
+pub struct AdmAdvertiserFilterSettings {
+    pub(crate) advertiser_hosts: Vec<String>,
+    pub(crate) impression_hosts: Vec<String>,
```

Comment:
Clicks & Impressions (`impression/click_hosts`) are _always_ going back to adM. Having a setting for a list of their valid hosts is a good idea but it shouldn't be per advertiser, it'd be global.

Whereas landing pages (`advertiser_url`/`advertiser_hosts`) take you to the advertiser itself, they need to be per advertiser as you already have it.

Reply:
I'm fine hard-coding the "filter" to take from `default` for those if you want to skip the partner level overrides.